src/main/java/io/sqooba/traildb/TrailDBIterator.java

```diff
@@ -52,6 +52,9 @@ public TrailDBEvent next() {
             public boolean hasNext() {
                 TrailDBEvent next = TrailDBNative.INSTANCE.cursorNext(TrailDBIterator.this.cursor);
                 if (next == null) {
+                    // Auto-close when at the end because it is not intuitive to have to close an
+                    // iterator when at its end in Java.
+                    TrailDBIterator.this.close();
```

Comment:
The issue was due to the fact I was not manually closing each iterator, I had the same problem when reading but forgot.
Yes when we have iterated over all events in the trail.